client/auth/AuthProvider.tsx

```diff
@@ -1,25 +1,47 @@
 import React from 'react';
+import { gql, useQuery } from '@apollo/client';
 
-interface User {
+interface UserContext {
     email: string;
     password: string;
     authenticated: boolean;
+    id: string;
 }
 
-const AuthContext = React.createContext<User | null>(null);
-
-// mock data
-// TODO: change this with real auth data
+type MyComponentProps = React.PropsWithChildren<unknown>;
```

Comment:
Kinda confused about this one. Why the need for type aliases?

Reply:
This is used on line 40 but I guess it doesn't need to be an alias. Maybe @feiw101 can comment.

Reply:
Refactored and removed global state